convert2rhel/unit_tests/utils_test.py

```diff
@@ -874,7 +888,312 @@ def communicate(self):
+        monkeypatch,
+    ):
+        process_mock = DummyPopenOutput(process_stdout, process_stderr)
+        output_mock = DummyPopenOutput(output_stdout, output_stderr, output_return_code)
+
+        monkeypatch.setattr(utils.subprocess, "Popen", mock.Mock(side_effect=[process_mock, output_mock]))
+        out, err, return_code = utils.run_piped_subprocess(cmd1, cmd2, print_error=True)
+
+        assert (out, err, return_code) == expected
+        assert "Error: %s" % err in caplog.records[-1].message
+
+    @pytest.mark.parametrize(
+        (
+            "cmd1",
+            "cmd2",
+            "process_stdout",
+            "process_stderr",
+            "output_stdout",
+            "output_stderr",
+            "output_return_code",
+            "skip_stdout_decode",
+            "expected",
+        ),
+        (
+            (
+                ["echo", "something", "'hi'"],
+                ["cpio", "-i", "--to-stdout"],
+                [u"test".encode()],
+                [u"".encode()],
+                [u"test".encode()],
+                [u"".encode()],
+                0,
+                True,
+                (b"test", "", 0),
+            ),
+            (
+                ["echo", "something", "'hi'"],
+                ["cpio", "-i", "--to-stdout"],
+                [u"testê café".encode("utf-8")],
+                [u"".encode("utf-8")],
+                [u"testê café".encode("utf-8")],
+                [u"".encode("utf-8")],
+                0,
+                False,
+                (u"testê café", "", 0),
+            ),
+        ),
+    )
+    def test_run_piped_subprocess_skip_stdout_decode(
+        self,
+        cmd1,
+        cmd2,
+        process_stdout,
+        process_stderr,
+        output_stdout,
+        output_stderr,
+        output_return_code,
+        skip_stdout_decode,
+        expected,
+        caplog,
+        monkeypatch,
+    ):
+        process_mock = DummyPopenOutput(process_stdout, process_stderr)
+        output_mock = DummyPopenOutput(output_stdout, output_stderr, output_return_code)
+
+        monkeypatch.setattr(utils.subprocess, "Popen", mock.Mock(side_effect=[process_mock, output_mock]))
+        out, err, return_code = utils.run_piped_subprocess(cmd1, cmd2, skip_stdout_decode=skip_stdout_decode)
+
+        assert (out, err, return_code) == expected
+
+    def test_run_piped_subprocess_with_devnull(self, monkeypatch):
+        cmd1 = ["echo", "something", "'hi'"]
+        cmd2 = ["cpio", "-i", "--to-stdout"]
+
+        process_mock = DummyPopenOutput([u"test".encode()], [u"".encode()])
+        output_mock = DummyPopenOutput([u"test".encode()], [u"".encode()], 0)
+
+        monkeypatch.setattr(utils.subprocess, "Popen", mock.Mock(side_effect=[process_mock, output_mock]))
+
+        with open(os.devnull, "w") as devnull:
+            out, err, return_code = utils.run_piped_subprocess(cmd1, cmd2, stdout=devnull)
+
+            assert (out, err, return_code) == ("", "", 0)
```

Comment:
I think we want to assert that the output did not get printed in this case. You can do that using the `capsysbnary` fixture: https://docs.pytest.org/en/7.1.x/how-to/capture-stdout-stderr.html#accessing-captured-output-from-a-test-function

Reply:
Not resolving this comment, but we have this mapped under the gist mentioned in a comment here, so should be fine to resolve it anyway.